### PR TITLE
Count: refactor tests to `@testing-library`

### PR DESCRIPTION
**Pull request by flootr**
#### Changes proposed in this Pull Request

* Count: refactor tests to `@testing-library`

#### Testing instructions

* Verify unit tests still pass
* Go to `/devdocs/design/count` and verify it still works as expected

Related to #63409

### CODE REVIEW

File: client/components/count/index.jsx [hunk 1]
```diff
@@ -30,6 +27,7 @@ Count.propTypes = {
 Count.defaultProps = {
 	primary: false,
 	compact: false,
+	numberFormat: ( i ) => i,
```
Comment (tyxla) [inline client/components/count/index.jsx]: 🤔 Where does the `numberFormat` come from, now that we removed the `localize` HoC?

Comment (flootr) [reply client/components/count/index.jsx]: Oops, you're right. I wasn't aware `numberFormat` is coming from `i18n-calypso`. Fixed via https://github.com/Automattic/wp-calypso/pull/64028/commits/b3d4055e199668d2c552784dd99ca23ae2d90337

File: client/components/count/test/index.jsx [hunk 2]
```diff
@@ -1,15 +1,23 @@
...
 	test( 'should call provided as prop numberFormat function', () => {
 		const numberFormatSpy = jest.fn();
-		shallow( <Count count={ 23 } numberFormat={ numberFormatSpy } /> );
+		render( <Count count={ 23 } numberFormat={ numberFormatSpy } /> );
 		expect( numberFormatSpy ).toHaveBeenCalledWith( 23 );
 	} );
+
+	test( 'should call `formatNumberCompact` if `compact` prop is `true`', () => {
+		const { container } = render( <Count count={ 1000 } compact /> );
+		expect( container.firstChild ).toHaveTextContent( '1.0K' );
```
Comment (tyxla) [inline client/components/count/test/index.jsx:16]: This test seems to be testing the internals of `i18n-calypso`'s `numberFormat()` so it likely doesn't belong here, but it's not a big deal I suppose.

**Review by tyxla — APPROVED** [review]
LGTM 👍 One little note, but it's totally optional and up to you.

**Pull request merged**